src/pkg/selectors/selectors.go

```diff
@@ -119,6 +121,33 @@ func (s Selector) String() string {
 	return string(bs)
 }
 
+// folderRefToPath takes a string that may be a ShortRef and returns a path.Path
+// if it matches a folder in deets. If ref does not match the pattern for
+// a ShortRef or it does not match any ShortRef in deets returns an error.
+func folderRefToPath(ref string, deets *details.Details) (path.Path, error) {
```

Comment:
Not sure I agree on the return values here.  Two reasons: 1/ an error implies that the caller either has some control over the input (and thus can fix it) or is required to back out of processing when the error arises.  The former isn't true (whichever controller calls this func is just feeding it end-user data), and the latter isn't wanted (no reason to back out of the operation here).  I think we can drop the error case, as it doesn't aid processing here.  But then we'll also want point 2.

2/ Returning a string, instead of a path, is more resilient for this case.  Consider that end users can only provide one of two inputs: a short ref, or an actual folder name/path.  If the input is not a ref, it must be an actual name/path.  By returning a string you can return either the matched path.Folder(), or the original string input, and the caller should be able to use the result without knowledge of the presence or absence of transformation (primarily because we're only doing prefix matching, which is okay).

There is a corner case here, which is: what if the user provides a valid shortRef string that matches no details entries?  Currently this returns a "NotFound" error.  In that case, according to the above suggestion, we'd return the original input again (the short ref), and it will (probably) match nothing in the reduction step, causing that selector to be ignored.  This is the same handling that occurs if the user provides, say, a typo ("inboz"), or other non-matching input.  By deferring to the reducer in this way, it also resolves another corner case in the current implementation: that an input can pass maybeShortRef without being a short ref, but will currently produce a "not found" error.

Reply:
> 1/ an error implies that the caller either has some control over the input (and thus can fix it) or is required to back out of processing when the error arises

I think an error can help shed light on the specific sort of failure that occurred, not just imply caller control or change in execution flow. However, given that you point out a corner case later on, I think it would be alright to have a single return value if that fit better overall

> 2/ Returning a string, instead of a path

Yea, I thought about returning a string too. This function is internal to the package though so the exported functions that call it can choose to munge the return values. The reason it's returning `path.Builder` instead of `string` is that OneDrive will need to remove the first three elements of the path. That's best done with some functions from the `path` package because it's already split the path. Returning a string would require re-parsing to the `path` package to remove leading elements

Reply:
> OneDrive will need to remove the first three elements of the path
Hmm, fair, fair.  Will be an interesting condition to comply with.